src/Features/LanguageServer/CommonLanguageServerProtocol.Framework/ILspLogger.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace CommonLanguageServerProtocol.Framework;
+
+public interface ILspLogger
```

Comment:
I believe that Roslyn's ILspLogger is custom. I'm not totally happy with this abstraction because it assumes that your logger implementation doesn't take parameters, so if it does you lose out on all the work I did in Razor to take advantage of those lol.

Reply:
Ya that was a little bit of my concern too. And if you wanted to take parameters you're then allocating extra strings for logging purposes. Definitely one of the benefits of MS.Extensions.Logging goodness. @dibarbet have you all not needed to pass extra information in your logger infra or have you just assumed the fire hose was on for logging and figured you'd be allocating anyhow?